jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpCookie.java

```diff
@@ -367,12 +367,8 @@ else if (version > 1)
             else
                 DateGenerator.formatCookieDate(buf, System.currentTimeMillis() + 1000L * _maxAge);
 
-            // for v1 cookies, also send max-age
-            if (version >= 1)
-            {
-                buf.append(";Max-Age=");
-                buf.append(_maxAge);
-            }
+            buf.append(";Max-Age=");
```

Comment:
IIRC, `Max-Age` shouldn't be sent along with `Expires` on the same cookie.
And starting with RFC6265 the `Expires` attribute is preferred.

Reply:
@joakime if you look at the code in eg jetty-11, we _have_ been sending `max-age` along with `expires` on RFC2965 cookies:  https://github.com/eclipse/jetty.project/blob/jetty-11.0.x/jetty-http/src/main/java/org/eclipse/jetty/http/HttpCookie.java#L355

As version is no longer meaningful for the `jakarta.servlet.http.Cookie` class I didn't include the test, but I can add it back in if you like?

Reply:
In Jetty 11 the `Version` field is only generated when the cookie.version is 1.
It defaults to 0.

And the `;Version=` field only exists for REALLY early Mozilla cookies.
IMO, we should drop `Max-Age` entirely, we have `Expires` which is supported in RFC2965 and RFC6265.

Cookies, and what they say, has gotten far more strict over the years (you can thank Firefox and Google for this).
Examples:

* Both browsers ignore the entire `Set-Cookie` line if it has any `$` (prefixed) attributes.
* Both browsers ignore `Set-Cookie` that attempts to set a cookie from a non-secure connection (if the cookie was previously set from the secure location, and this has nothing to do with any of the secure cookie attributes)
* Both browsers ignore the entire `Set-Cookie` line if it has a mangled quoting or escaping (this includes improperly escaped content within the double quotes, or bad pct-encoding)
* Both browsers ignore the entire `Set-Cookie` line if it has a `Version` attribute (of any value)
* The default CORS behaviors are getting more strict (this is the mode that the browser operates in if no CORS headers are sent)
* Chrome ignores `Set-Cookie` on 302 redirects.
* Both browsers ignore the entire `Set-Cookie` line if `SameSite` is set poorly.
* Chrome ignores the `Set-Cookie` line if `SameSite=None` and the `Secure` attribute is missing. (which automatically means you cannot use `SameSite=None` on a normal HTTP connection)
* Both browsers currently reject the entire `Set-Cookie` line if [cookie prefixes](https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-cookie-prefixes-00) are used in conjunction with forbidden common attributes. (eg: `__Host` prefix with `Domain` attribute is forbidden)
* Both browsers currently throw warnings if `Max-Age` and `Expires` are both present on a `Set-Cookie` line (despite what the RFC says) (they are evaluating what to do, both seem to indicate that they will just ignore `Max-Age` in this situation in the future)

Reply:
@joakime bottom line is the servlet tck is testing for `Max-Age` being sent:   https://github.com/jakartaee/platform-tck/blob/master/src/com/sun/ts/tests/servlet/api/jakarta_servlet_http/cookie/URLClient.java#L368